nntrainer/layers/layer_impl.cpp

```diff
@@ -23,7 +24,7 @@ namespace nntrainer {
 
 LayerImpl::LayerImpl() :
   finalized(false),
-  layer_impl_props(std::make_unique<std::tuple<>>()) {}
```

Comment:
props trainable will be in LayerNode and not in LayerImpl, and will be accessible to Layer via Init/Run context.

Reply:
Thought about it, there are some issue

1. What if a layer is not trainable by default, or must be always non trainable?
2. When we request tensor, only layer knows which layer should be trainable or non-trainable
    eg) Even if we set trainable, weight1 and weight2 is trainable but weight3, weight4 must stay non-trainble
          Even if we set non-trainable, weight1 must be trainable.

Reply:
My thinking is that layer has the boolean trainable but when request weight set true or false respectively.

Reply:
> Thought about it, there are some issue
> 
> 1. What if a layer is not trainable by default, or must be always non trainable?

yeah, there are pros and cons. To support `always non-trainable`, there will yet another function in layer devel like `supportTraining()` if trainable in LayerNode. If in LayerImpl, we need `getTrainable` or `getProperty` now. Both fine choices, left up to you.

> 2. When we request tensor, only layer knows which layer should be trainable or non-trainable
>    eg) Even if we set trainable, weight1 and weight2 is trainable but weight3, weight4 must stay non-trainble
>    Even if we set non-trainable, weight1 must be trainable.

yes, that must be controlled by the layer. When layer requestWeights() by context, layer gets full control of this. This true/false for each weight must be `and` with the trainable of the layer, and then correspondingly set to weight (not yet done, just my thought).

Reply:
If you choose to keep trainable in LayerImpl, please update LayerNode correspondingly as well (remove trainable from there).

Reply:
Thanks for the suggestion, I'll update within this PR.

Could you also take a look over #1297 regarding this?

Reply:
done.

Reply:
> I'll update within this PR.

trainable deleted from props

Reply:
>When we request tensor, only layer knows which layer should be trainable or non-trainable
eg) Even if we set trainable, weight1 and weight2 is trainable but weight3, weight4 must stay non-trainble
Even if we set non-trainable, weight1 must be trainable.

I wonder whether we are going to set the trainable in weight ??

Reply:
Yes, we are setting trainable in weights and that won't be changing.